src/main/java/org/opensearch/commons/destination/message/LegacyBaseMessage.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package org.opensearch.commons.destination.message;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Map;
+
+import org.apache.http.client.utils.URIBuilder;
+import org.opensearch.common.Strings;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.io.stream.Writeable;
+
+/**
+ * This class holds the generic parameters required for a
+ * message.
+ */
+public abstract class LegacyBaseMessage implements Writeable {
+
+    protected LegacyDestinationType destinationType;
+    protected String destinationName;
+    protected String url;
+    private String content;
+
+    LegacyBaseMessage(final LegacyDestinationType destinationType, final String destinationName, final String content) {
+        if (destinationType == null) {
+            throw new IllegalArgumentException("Channel type must be defined");
+        }
+        if (!Strings.hasLength(destinationName)) {
+            throw new IllegalArgumentException("Channel name must be defined");
+        }
+        this.destinationType = destinationType;
+        this.destinationName = destinationName;
+        this.content = content;
+    }
+
+    LegacyBaseMessage(final LegacyDestinationType destinationType, final String destinationName, final String content, final String url) {
+        this(destinationType, destinationName, content);
+        if (url == null) {
+            throw new IllegalArgumentException("url is invalid or empty");
+        }
+        this.url = url;
```

Comment:
Please validate that schema is https. http schema is not supported.

Reply:
http must be supported as alerting and ism have already supported it in production